src/hooks/useTodoCards.ts

```diff
@@ -0,0 +1,35 @@
+import { useState, useEffect } from "react"
+import { ITodoItemCard } from "../components/todo-item-card/ITodoItemCard";
+import axios, { AxiosError } from "axios";
+
+const useTodoCards = () => {
+  const [todoCards, setTodoCards] = useState<ITodoItemCard[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState("");
+
+  function cardHook(todoItemCard: ITodoItemCard) {
+    setTodoCards(prev => [...prev, todoItemCard]);
+  }
+
+  async function fetchCards() {
+    try {
+      setError("");
+      setIsLoading(true);
+      const response = await axios.get<ITodoItemCard[]>("https://aufgabenliste.azurewebsites.net/api/todo")
+      setTodoCards(response.data);
+      setIsLoading(false);
+    } catch (e: unknown) {
+      const error = e as AxiosError;
+      setIsLoading(false);
+      setError(error.message);
+    }
+  }
+
+  useEffect(() => {
+    fetchCards();
+  }, []);
```

Comment:
We should discuss it...looks like a hack to automatically call the server when the hook is calling (Better to have control on that inside the component),

Reply:
If I understood correct, this trick for once render on load page. But anyway we need to discuss it.